frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/usePodStepsStates.ts

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { PodContainer } from '~/types';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { useDeepCompareMemoize } from '~/utilities/useDeepCompareMemoize';
+import { getPodStepsStates } from '~/concepts/k8s/pods/utils';
+
+const usePodStepsStates = (
+  podContainers: PodContainer[],
+  podName: string,
+): {
+  stepName: string;
+  state: string;
+}[] => {
+  const { namespace } = usePipelinesAPI();
+  const podSteps = useDeepCompareMemoize(podContainers);
+  const [allpodStepsState, setAllPodStepsState] = React.useState<
+    { stepName: string; state: string }[]
+  >([]);
+
+  React.useEffect(() => {
+    getPodStepsStates(podSteps, namespace, podName)
```

Comment:
nit: In `getPodStepsStates`, all promises are awaited for and the results are returned from that function. Since we're awaiting promise results again here, I'm wondering if maybe that function should just return `Promise.all(...)` based on the usage here.